pkg/proxy/conntrack/conntrack.go

```diff
@@ -49,95 +50,143 @@ type Interface interface {
 		// These stale udp connection will keep black hole traffic. Making this a best effort operation for now, since it
 		// is expensive to baby-sit all udp connections to kubernetes services.
-		return fmt.Errorf("error deleting connection tracking state for UDP service IP: %s, error: %v", ip, err)
+		return fmt.Errorf("error deleting connection tracking state for %s service IP: %s, error: %w", protoStr(protocol), ip, err)
 	}
+	klog.V(4).InfoS("Cleared conntrack entries", "count", n)
 	return nil
 }
 
-// ClearEntriesForPort is part of Interface
-func (ct *execCT) ClearEntriesForPort(port int, isIPv6 bool, protocol v1.Protocol) error {
+// ClearEntriesForPort delete the conntrack entries for connections specified by the
+// destination Port(original direction) and IPFamily.
+func (ct *conntracker) ClearEntriesForPort(port int, isIPv6 bool, protocol v1.Protocol) error {
+	filter := &conntrackFilter{
+		protocol: getProtocolNumber(protocol),
+		original: &connectionTuple{
+			dstPort: uint16(port),
+		},
+	}
 	if port <= 0 {
 		return fmt.Errorf("wrong port number. The port number must be greater than zero")
 	}
-	parameters := parametersWithFamily(isIPv6, "-D", "-p", protoStr(protocol), "--dport", strconv.Itoa(port))
-	err := ct.exec(parameters...)
-	if err != nil && !strings.Contains(err.Error(), noConnectionToDelete) {
-		return fmt.Errorf("error deleting conntrack entries for UDP port: %d, error: %v", port, err)
+
+	klog.V(4).InfoS("Clearing conntrack entries", "port", port, "protocol", protocol)
+	n, err := ct.handler.ConntrackDeleteFilters(netlink.ConntrackTable, getNetlinkFamily(isIPv6), filter)
+	if err != nil {
+		return fmt.Errorf("error deleting connection tracking state for %s port: %d, error: %w", protoStr(protocol), port, err)
+	}
+	if n == 0 {
+		return fmt.Errorf("error deleting connection tracking state for %s port: %d, no entries found", protoStr(protocol), port)
 	}
 	return nil
 }
 
-// ClearEntriesForNAT is part of Interface
-func (ct *execCT) ClearEntriesForNAT(origin, dest string, protocol v1.Protocol) error {
-	parameters := parametersWithFamily(utilnet.IsIPv6String(origin), "-D", "--orig-dst", origin, "--dst-nat", dest,
-		"-p", protoStr(protocol))
-	err := ct.exec(parameters...)
-	if err != nil && !strings.Contains(err.Error(), noConnectionToDelete) {
+// ClearEntriesForNAT delete the conntrack entries for connections specified by the
+// destination IP(original direction) and source IP(reply direction).
+func (ct *conntracker) ClearEntriesForNAT(origin, dest string, protocol v1.Protocol) error {
+	filter := &conntrackFilter{
+		protocol: getProtocolNumber(protocol),
+		original: &connectionTuple{
+			dstIP: netutils.ParseIPSloppy(origin),
+		},
+		reply: &connectionTuple{
+			srcIP: netutils.ParseIPSloppy(dest),
+		},
+	}
+
+	klog.V(4).InfoS("Clearing conntrack entries", "origin", origin, "destination", dest, "protocol", protocol)
```

Comment:
destination nat? is not really the destination, is the natted ip

Reply:
The godoc on the interface method describes the terms 'origin' and 'dest' and I kept the logging as before only.

https://github.com/kubernetes/kubernetes/blob/e9c9a27c97634c433dd9dad1118a98b7e4d519f8/pkg/proxy/conntrack/conntrack.go#L43-L45

Reply:
Should I change the arguments as well, both in the interface definition and implementation?

Reply:
I think arguments are ok, are variables, but logging will be confusing for troubleshooting

Reply:
we may use the same convention than the man page https://manpages.ubuntu.com/manpages/focal/en/man8/conntrack.8.html

`orig-src` y `reply-src` so the logs and the conntrack entries meanings are the same

Reply:
cool, will update with `orig-dst` and `reply-src`.